tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -118,13 +119,18 @@ def images(self) -> Dict[str, List[VmImageInfo]]:
         """
         return self.__images
 
+    RANDOM_IMAGES_RE = re.compile(r"random\((?P<image>[^,]+)(\s*,\s*(?P<count>\d+))?\)")
+
     def _validate(self):
         """
         Performs some basic validations on the data loaded from the YAML description files
         """
         for suite in self.test_suites:
             # Validate that the images the suite must run on are in images.yml
             for image in suite.images:
+                match = AgentTestLoader.RANDOM_IMAGES_RE.match(image)
+                if match is not None:
+                    image = match.group('image')
```

Comment:
does match.group('image') give us <image_set> that we parsed in the regular expression?

I'm wondering if this would be more readable if we changed < image > to <image_set> in RANDOM_IMAGES_RE.

Leaving it as < image > is a bit confusing to read since we're also using image for the for loop here